app/Support/Shortcode/Shortcode.php

```diff
@@ -171,12 +171,40 @@ private function parse(string $input, array $options = []): string
 
     private function renderUrlLink(ShortcodeInterface $shortcode): string
     {
-        return '<a href="' . $this->protocolPrefix($shortcode->getBbCode() ?: $shortcode->getContent()) . '">' . ($shortcode->getContent() ?: $this->protocolPrefix($shortcode->getBbCode())) . '</a>';
+        // Determine the URL from the shortcode's BBCode or content.
+        $url = $shortcode->getBbCode() ?: $shortcode->getContent();
+
+        // Ensure the correct protocol prefix (http/https) is being used.
+        $prefixedUrl = $this->protocolPrefix($url);
+
+        // Is this an external link? If so, it needs to go through the redirect gateway.
+        $finalUrl = $prefixedUrl;
+        if (strpos($prefixedUrl, 'retroachievements.org') === false) {
+            $finalUrl = route('redirect', ['url' => $prefixedUrl]);
+        }
+
+        // Finally, build and return the needed anchor tag.
+        // HTML escape the visible URL to prevent XSS vulnerabilities.
+        return sprintf('<a href="%s">%s</a>', $finalUrl, htmlspecialchars($prefixedUrl, ENT_QUOTES, 'UTF-8'));
```

Comment:
This is no longer using the text for the link:
```
Go to [url=http://google.com]Google[/url] then [url=http://retroachievements.org/]RetroAchievements[/url]!
```
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/0e753e90-30d3-4880-95bb-c4176530a153)

instead of
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/bfc40d99-a19e-482c-b305-584b9b7331fd)

Reply:
Resolved in latest.